controllers/updates/index.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+const path = require('path');
+const express = require('express');
+const { concat, isArray, pick, get } = require('lodash');
+
+const router = express.Router();
+
+const { injectBreadcrumbs } = require('../../middleware/inject-content');
+const { getUpdates } = require('../../services/content-api');
+const appData = require('../../modules/appData');
+const { buildPagination } = require('../../modules/pagination');
+
+function renderListing({ res, title, entries = [], entriesMeta = null, breadcrumbs = [], isNewsLandingPage = false }) {
+    const pagination = buildPagination(entriesMeta.pagination);
+    res.render(path.resolve(__dirname, './views/listing'), {
+        title,
+        entries,
+        entriesMeta,
+        pagination,
+        breadcrumbs,
+        isNewsLandingPage
+    });
+}
+
+if (appData.isNotProduction) {
+    router.get('/:type?/:date?/:slug?', injectBreadcrumbs, async (req, res, next) => {
+        let urlPath = '';
+        let query = {};
+
+        const isSinglePost = req.params.date && req.params.slug;
+
+        if (req.params.type) {
+            // Due to quirks of Craft's entry types,
+            // we have to convert snake_case into kebab-case
+            // both at the API end and then in reverse here.
+            urlPath += `/${req.params.type.replace('-', '_')}`;
+        }
+
+        if (isSinglePost) {
+            urlPath += `/${req.params.date}/${req.params.slug}`;
+        } else {
+            // Listings can be filtered by the following valid query parameters
+            query = pick(req.query, ['page', 'tag', 'author', 'category']);
+        }
+
+        try {
+            const response = await getUpdates({
+                locale: req.i18n.getLocale(),
+                urlPath: urlPath,
+                query: query
+            });
+
+            if (response.result) {
+                // Do we have a listing page?
+                if (isArray(response.result)) {
+                    // @TODO i18n for 'All news' and pageTitles below
+                    // Is this a specific listing page (eg. blog, press releases) or everything?
+                    const filterType = get(response, 'meta.pageType');
+                    const isNewsLandingPage = !req.params.type && filterType === 'single';
+                    // @TODO these titles are singular - should they be pluralised here manually or in the CMS?
```

Comment:
Really not sure how best to handle this – we could map the types here (guess we'll need to translate them) or translate them in Craft?

Reply:
Yeah agreed – have added singular/plural translations so we can do this sort of thing:

![image](https://user-images.githubusercontent.com/394376/48485597-30771a00-e811-11e8-8938-ea5c950f7fbe.png)

![image](https://user-images.githubusercontent.com/394376/48485618-38cf5500-e811-11e8-94f4-5f1f41cee4fc.png)